examples/platform/silabs/SilabsDeviceDataProvider.cpp

```diff
@@ -339,7 +348,16 @@ CHIP_ERROR SilabsDeviceDataProvider::GetVendorId(uint16_t & vendorId)
 CHIP_ERROR SilabsDeviceDataProvider::GetProductName(char * buf, size_t bufSize)
 {
     size_t productNameLen = 0; // without counting null-terminator
-    return SilabsConfig::ReadConfigValueStr(SilabsConfig::kConfigKey_ProductName, buf, bufSize, productNameLen);
+    CHIP_ERROR err        = SilabsConfig::ReadConfigValueStr(SilabsConfig::kConfigKey_ProductName, buf, bufSize, productNameLen);
+#if defined(CHIP_DEVICE_CONFIG_TEST_PRODUCT_NAME)
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        VerifyOrReturnError(buf != nullptr, CHIP_ERROR_NO_MEMORY);
+        memcpy(buf, CHIP_DEVICE_CONFIG_TEST_PRODUCT_NAME, bufSize);
```

Comment:
Again, what guarantees `CHIP_DEVICE_CONFIG_TEST_PRODUCT_NAME` is at least `bufSize` bytes?